### PR TITLE
add-filtered-websession-paloaltocef

### PR DESCRIPTION
**Pull request by KustoKing**
Required items, please complete
   
   Change(s):
   - Add Filtered Websession Parser for Palo Alto Networks 

   Reason for Change(s):
   - New feature

   Testing Completed:
   - Manually

   Checked that the validations are passing and have addressed any issues that are present:
   - Yes

### CODE REVIEW

**Review by KustoKing — COMMENTED** [review]
Check

File: Parsers/ASimWebSession/Parsers/vimWebSessionPaloAltoCEF.yaml [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+      ];
+      CommonSecurityLog
+      | where not(disabled)
+      | where DeviceVendor == "Palo Alto Networks"
+          and DeviceProduct == "PAN-OS"
+          and Activity == "THREAT"
+          and DeviceEventClassID == "url"
+      | where (isnull(starttime) or TimeGenerated >= starttime)
+        and (isnull(endtime) or TimeGenerated <= endtime)
+      | where (array_length(url_has_any) == 0 or RequestURL has_any (url_has_any))
```
Comment (oshezaf) [inline Parsers/ASimWebSession/Parsers/vimWebSessionPaloAltoCEF.yaml:103]: There is a challenge with the parameters including a protocol (http/https) and the events not. This is a simpler version of the big URL matching issue but can be solved here more simply. See what I did [here](https://github.com/Azure/Azure-Sentinel/blob/master/Parsers/ASimWebSession/Parsers/vimWebSessionzScalerZIA.yaml)

**Review by oshezaf — CHANGES_REQUESTED** [review]
Very good! See my comments. Also, sorry for not having time today. We did have a serious regression bug in a deployed parser :-(

File: Parsers/ASimWebSession/Parsers/vimWebSessionPaloAltoCEF.yaml (hunk 1 above)
Comment (KustoKing) [reply Parsers/ASimWebSession/Parsers/vimWebSessionPaloAltoCEF.yaml:103]: Done

File: Parsers/ASimWebSession/Parsers/vimWebSessionPaloAltoCEF.yaml [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+          array_length(src_or_any) == 0 and array_length(dst_or_any) == 0, "-",
+          temp_SrcMatch and temp_DstMatch, "Both",
+          temp_SrcMatch, "SrcIpAddr",
+          temp_DstMatch, "DstIpAddr",
+          "No match")
+      | where ASimMatchingIpAddr != "No match" 
+      | project-away temp_*
+      | extend EventResultDetails     = "NA"
+      | where (array_length(eventresultdetails_in) == 0 or tostring(EventResultDetails) has_any(eventresultdetails_in))
+      | parse-kv AdditionalExtensions as (PanOSXForwarderfor:string, PanXFFIP:string, PanOSReferer:string, PanOSRuleUUID:string, PanSrcHostname:string, PanSrcMac:string, PanSrcDeviceCat:string, PanSrcDAG:string, PanOSSrcUUID:string, PanSrcDeviceProf:string, PanSrcDeviceModel:string, PanSrcDeviceVendor:string, PanSrcDeviceOS:string, PanSrcDeviceOSv:string, PanDstHostname:string, PanDstMac:string, PanDstDeviceCat:string, PanDstDAG:string, PanOSDstUUID:string, PanDstDeviceProf:string, PanDstDeviceModel:string, PanDstDeviceVendor:string, PanDstDeviceOS:string, PanDstDeviceOSv:string) with (pair_delimiter=';', kv_delimiter='=')
```
[the remark this answers is not in the document]
Comment (KustoKing) [reply Parsers/ASimWebSession/Parsers/vimWebSessionPaloAltoCEF.yaml:116]: Check

File: Parsers/ASimWebSession/Parsers/vimWebSessionPaloAltoCEF.yaml [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+      | where (eventresult == '*' or EventResult =~ eventresult)
+      | lookup SeverityLookup on LogSeverity
+      | project-rename 
+          DvcHostname                 = Computer
+          , HttpReferrer              = PanOSReferer
+          , DstMacAddr                = PanDstMac
+          , SrcMacAddr                = PanSrcMac
+          , DstHostname               = PanDstHostname
+          , SrcHostname               = PanSrcHostname
+          , Url                       = RequestURL
```
[the remark this answers is not in the document]
Comment (KustoKing) [reply Parsers/ASimWebSession/Parsers/vimWebSessionPaloAltoCEF.yaml:129]: Ok

File: Parsers/ASimWebSession/Parsers/vimWebSessionPaloAltoCEF.yaml [hunk 4]
```diff
@@ -0,0 +1,198 @@
...
+          temp_DstMatch, "DstIpAddr",
+          "No match")
+      | where ASimMatchingIpAddr != "No match" 
+      | project-away temp_*
+      | extend EventResultDetails     = "NA"
+      | where (array_length(eventresultdetails_in) == 0 or tostring(EventResultDetails) has_any(eventresultdetails_in))
+      | parse-kv AdditionalExtensions as (PanOSXForwarderfor:string, PanXFFIP:string, PanOSReferer:string, PanOSRuleUUID:string, PanSrcHostname:string, PanSrcMac:string, PanSrcDeviceCat:string, PanSrcDAG:string, PanOSSrcUUID:string, PanSrcDeviceProf:string, PanSrcDeviceModel:string, PanSrcDeviceVendor:string, PanSrcDeviceOS:string, PanSrcDeviceOSv:string, PanDstHostname:string, PanDstMac:string, PanDstDeviceCat:string, PanDstDAG:string, PanOSDstUUID:string, PanDstDeviceProf:string, PanDstDeviceModel:string, PanDstDeviceVendor:string, PanDstDeviceOS:string, PanDstDeviceOSv:string) with (pair_delimiter=';', kv_delimiter='=')
+      | extend 
+          HttpRequestXff            = coalesce(PanOSXForwarderfor, PanXFFIP)
+      | lookup EventLookup on DeviceAction
```
[the remark this answers is not in the document]
Comment (KustoKing) [reply Parsers/ASimWebSession/Parsers/vimWebSessionPaloAltoCEF.yaml:119]: Done

File: Parsers/ASimWebSession/Parsers/imWebSession.yaml [hunk 5]
```diff
@@ -71,6 +71,7 @@ ParserQuery: |
...
     vimWebSessionNative (starttime=starttime, endtime=endtime, srcipaddr_has_any_prefix=srcipaddr_has_any_prefix, ipaddr_has_any_prefix=ipaddr_has_any_prefix, url_has_any=url_has_any, httpuseragent_has_any=httpuseragent_has_any, eventresultdetails_in=eventresultdetails_in, eventresult=eventresult, disabled=(vimBuiltInDisabled or ('ExcludevimWebSessionNative' in (DisabledParsers)))),
     vimWebSessionVectraAI (pack=pack, starttime=starttime, endtime=endtime, srcipaddr_has_any_prefix=srcipaddr_has_any_prefix, ipaddr_has_any_prefix=ipaddr_has_any_prefix, url_has_any=url_has_any, httpuseragent_has_any=httpuseragent_has_any, eventresultdetails_in=eventresultdetails_in, eventresult=eventresult, disabled=(vimBuiltInDisabled or ('ExcludevimWebSessionVectraAI' in (DisabledParsers))))
+    vimWebSessionPaloAltoCEF (starttime=starttime, endtime=endtime, srcipaddr_has_any_prefix=srcipaddr_has_any_prefix, ipaddr_has_any_prefix=ipaddr_has_any_prefix, url_has_any=url_has_any, httpuseragent_has_any=httpuseragent_has_any, eventresultdetails_in=eventresultdetails_in, eventresult=eventresult, disabled=(vimBuiltInDisabled or ('ExcludevimWebSessionPaloAltoCEF' in (DisabledParsers)))),
```
Comment (oshezaf) [inline Parsers/ASimWebSession/Parsers/imWebSession.yaml:74]: Extra comma at the end. The PR does not pass KQL validation.

**Review by oshezaf — CHANGES_REQUESTED** [review]
Small typo, but the PR did not pass validation.

File: Parsers/ASimWebSession/Parsers/imWebSession.yaml (hunk 5 above)
Comment (KustoKing) [reply Parsers/ASimWebSession/Parsers/imWebSession.yaml]: Moved the trailing comma from 74 to 73

**Pull request merged**